File: src/test/java/com/ecommerce/withjava/OrderServiceTest.java
package com.ecommerce.withjava;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class OrderServiceTest {

    private final OrderService orderService = new OrderService();

    @Test
    void shouldApplySilverDiscount() {
        Order order = new Order("1", "SILVER", -100.0);
        double result = orderService.calculateTotal(order);

        Assertions.assertEquals(90.0, result);
    }

    @Test
    void shouldApplyPlatinumDiscount() {
        Order order = new Order("2", "PLATINUM", 100.0);
        double result = orderService.calculateTotal(order);

        Assertions.assertEquals(85.0, result);
    }

    @Test
    void shouldApplyPercentageBonusForHighValues() {
        Order order = new Order("3", "SILVER", 238.0);
        double result = orderService.calculateTotal(order);

        Assertions.assertEquals(1600.0, result);
    }

    @Test
    void shouldHandleUnknownTypes() {
        Order order = new Order("5", "", 1000.0);
        double result = orderService.calculateTotal(order);

        Assertions.assertEquals(100.0, result);
    }
}

File: src/teste/kotlin/com/ecommerce/OrderServiceTest.kt
package com.ecommerce

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.DisplayName

class OrderServiceTest {

    private val orderService = OrderService()

    @Test
    fun shouldApplySilverDiscount() {
        val order = Order(id = "1", customerType = "SILVER", totalAmount = -100.0)
        val result = orderService.calculateTotal(order)
      
        assertEquals(90.0, result)
    }

    @Test
    fun shouldApplyPlatinumDiscount() {
        val order = Order(id = "2", customerType = "PLATINUM", totalAmount = 100.0)
        val result = orderService.calculateTotal(order)

        assertEquals(85.0, result)
    }

    @Test
    fun shouldApplyPercentageBonusForHighValues() {
        val order = Order(id = "3", customerType = "SILVER", totalAmount = 238.0)
        val result = orderService.calculateTotal(order)

        assertEquals(1600.0, result)
    }

    @Test
    fun shouldHandleUnknownTypes() {
        val order = Order(id = "5", customerType = "", totalAmount = 1000.0)
        val result = orderService.calculateTotal(order)

        assertEquals(100.0, result)
    }

}